view_generator/main.tf

```diff
@@ -0,0 +1,60 @@
+/**
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// HACK -- should use local schema JSON instead of fetching it
+//    but right now we're waiting for tables to instantiate
+resource "null_resource" "before" {
+}
+
+resource "null_resource" "delay" {
+  provisioner "local-exec" {
+    command = "sleep 10"
+  }
+  triggers = {
+    "before" = "${null_resource.before.id}"
+  }
+}
+
+resource "null_resource" "main" {
+  depends_on = ["null_resource.delay"]
+  count      = length(var.authorized_views)
+
+  triggers = {
+    _always = uuid()
+  }
+
+  provisioner "local-exec" {
```

Comment:
I'm confused why we need to use local-exec? It seems the Terraform resource natively supports `view` resources: https://www.terraform.io/docs/providers/google/r/bigquery_table.html#view

Reply:
I saw that, but I think the module was created this way because it allows for creating multiple authorized views for the same table. In the table TF resource, you can only create one view for the one table.

So if you want this module to take the approach that 1 table == 1 view, then we can use that TF resource.

Reply:
Let's keep it as-is for now and open an issue (on GitHub) to re-assess later.